src/shared.rs

```diff
@@ -16,43 +17,81 @@ pub(crate) const STREAM_MAP: &str = "stream-map";
 pub(crate) const BATCH_MAP: &str = "batch-map";
 const MINIMUM_NUMAFLOW_VERSION: &str = "1.2.0-rc4";
 
-// default_info_file is a function to get a default server info json
-// file content. This is used to write the server info file.
-// This function is used in the write_info_file function.
-// This function is not exposed to the user.
-pub fn default_info_file() -> serde_json::Value {
-    let metadata: HashMap<String, String> = HashMap::new();
-    serde_json::json!({
-        "protocol": "uds",
-        "language": "rust",
-        "version": "0.0.1",
-        "metadata": metadata,
-        "minimum-numaflow-version": MINIMUM_NUMAFLOW_VERSION,
-    })
+// ServerInfo structure to store server-related information
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct ServerInfo {
+    #[serde(default)]
+    protocol: String,
+    #[serde(default)]
+    language: String,
+    #[serde(default)]
+    minimum_numaflow_version: String,
+    #[serde(default)]
+    version: String,
+    #[serde(default)]
+    metadata: Option<HashMap<String, String>>, // Metadata is optional
+}
+impl ServerInfo {
+    // default_info_file is a function to get a default server info json
+    // file content. This is used to write the server info file.
+    // This function is used in the write_info_file function.
+    // This function is not exposed to the user.
+    pub fn default() -> Self {
+        let metadata: HashMap<String, String> = HashMap::new();
+        // Return the default server info json content
+        // Create a ServerInfo object with default values
+        ServerInfo {
+            protocol: "uds".to_string(),
+            language: "rust".to_string(),
+            minimum_numaflow_version: MINIMUM_NUMAFLOW_VERSION.to_string(),
```

Comment:
the default version should be `0.0.0-pre0`?

Reply:
For minimum numaflow version?

Reply:
my bad.. we need to manually update.